src/main/java/edu/ucla/library/libcal/handlers/ProxyHandler.java

```diff
@@ -91,20 +99,29 @@ public ProxyHandler(final Vertx aVertx, final JsonObject aConfig) {
     public void handle(final RoutingContext aContext) {
         final HttpServerResponse response = aContext.response();
         final String path = aContext.request().path();
-        final String method = aContext.request().method().name();
+        final HttpMethod method = aContext.request().method(); // .name();
+        final RequestBody payload = aContext.body();
         final String originalClientIP = aContext.request().remoteAddress().hostAddress();
         final Cidr4Trie<String> allowedIPs = buildAllowedNetwork(myConfig.getString(Config.ALLOWED_IPS).split(COMMA));
 
         if (isOnNetwork(new Ip4(originalClientIP), allowedIPs)) {
             final String receivedQuery = path.concat(
                     aContext.request().query() != null ? QUESTION_MARK.concat(aContext.request().query()) : EMPTY);
             myTokenProxy.getBearerToken().compose(token -> {
-                if (GET.equals(method)) {
-                    return myApiProxy.getLibCalOutput(token, receivedQuery).map(myMapper::decode);
-                } else {
-                    return myApiProxy.postLibCalOutput(token, receivedQuery, aContext.body().asJsonObject())
-                            .map(myMapper::decode);
-                }
+		    try {
+                return myApiProxy.getLibCalOutput(token, receivedQuery, method,
+                        (payload != null ? payload.asJsonObject() : null)).map(myMapper::decode);
+		    } catch (Exception details) {
+		    	details.printStackTrace();
```

Comment:
Log errors rather than writing to stdout. And the line below's indentation is off.

Reply:
Ditto.